### PR TITLE
XW-1442 create tests for checking if title on category page is set

### PR DESCRIPTION
**Pull request by bognix**
Merge after https://github.com/Wikia/mercury/pull/2416 is released

@rogatty

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/CategoryPageTest.java [hunk 1]
```diff
@@ -45,10 +48,25 @@ public void mercury_category_navigateThroughSection() {
...
         .navigateToPageWithArticleAndWithMembersFromUrl()
         .navigateToCategoryMemberPage("Category test 001");
   }
+
+  @Test(
+          groups = "mercury_category_title_present",
+          dataProvider = "getCategoriesWithTitles",
+          dataProviderClass = MercuryCategoriesDataProvider.class
+  )
+  public void mercury_category_title_present(String pageUrl, String expectedTitle) {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/CategoryPageTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/common/dataprovider/MercuryCategoriesDataProvider.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.wikia.webdriver.common.dataprovider;
+
+import org.testng.annotations.DataProvider;
+
+public class MercuryCategoriesDataProvider {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/dataprovider/MercuryCategoriesDataProvider.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1118)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by rogatty** [comment]
LGTM

File: src/test/java/com/wikia/webdriver/common/dataprovider/MercuryCategoriesDataProvider.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    @DataProvider
+    public static final Object[][] getCategoriesWithTitles() {
+        return new Object[][]{
+                {"/wiki/Category:NoContent", "NOCONTENT"},
+                {"/wiki/Category:WithContent", "WITHCONTENT"},
+                {"/wiki/Category:DisplayTitleWithContent", "CUSTOM DISPLAY TITLE WITH CONTENT"}
+                //Below case can be enabled when XW-1461 is fixed
+                //{"/wiki/Category:DisplayTitleNoContent", "CUSTOM DISPLAY TITLE WITHOUT CONTENT"}
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/dataprovider/MercuryCategoriesDataProvider.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://sonar-s1/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/CategoryPageTest.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/CategoryPageTest.java]: MAJOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)

File: src/test/java/com/wikia/webdriver/common/dataprovider/MercuryCategoriesDataProvider.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package com.wikia.webdriver.common.dataprovider;
+
+import org.testng.annotations.DataProvider;
+
+public class MercuryCategoriesDataProvider {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/dataprovider/MercuryCategoriesDataProvider.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1118)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by Archbold** [comment]
@bognix  can you close this PR if no further work is being done, then reopen it when work starts again?

**Comment by tomasznapieralski** [comment]
IMO we can test titles and sublitles on lower levels: api tests, unit tests, there is no need for functional tests for that

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Pull request closed without merging**